### DISCUSSION TITLE
updateFormState is called (causes 4 times slower) when a new empty form is loading.

### DISCUSSION THREAD

**Discussion by equinix-sulai** [react-hook-form/react-hook-form#6887] [Q&A]
Hi All,

We have an app that benefited hugely from react-hook-form.
Thanks to all contributors to this project. 

Our app is using react-hook-form@6.11.5.

We found that if we use "onSubmit" mode (we use "all" mode with useForm now), the page load **4 times faster!** The profiling shows the "updateFormState" likely to be the reason. May I know is it possible to avoid the state update when we load an empty form regardless of the mode? 

useForm.ts (around line 1040)
 ```

      if (!isOnSubmit && readFormStateRef.current.isValid) {
        validateField(
          fieldsRef,
          isValidateAllFieldCriteria,
          field,
          shallowFieldsStateRef,
        ).then((error: FieldErrors) => {
          const previousFormIsValid = formStateRef.current.isValid;

          isEmptyObject(error)
            ? set(validFieldsRef.current, name, true)
            : unset(validFieldsRef.current, name);

          if (previousFormIsValid !== isEmptyObject(error)) {
            updateFormState();
          }
        });
  
```

Thanks.

**Comment by bluebill1049** [comment]
perhaps you should avoid `useForm` when the form is empty.

`!isEmpty && <WhateverFormHasUseForm />`

**Reply by equinix-sulai** [reply]
Thanks. However, even the form is empty, we still do registration, watch (and other methods from useForm) for each input field when the page/form is loading.

Just like to elaborate that the "new empty form" means a page with inputs that are not "touched" and not "dirty".

**Comment by equinix-sulai** [comment]
@bluebill1049 appreciated if you have any suggestions to work around the above issue? thanks.
